python/tests/test_account.py

```diff
@@ -1077,6 +1077,44 @@ def test_send_receive_locations(self, acfactory, lp):
         locations3 = chat2.get_locations(contact=contact)
         assert not locations3
 
+    def test_no_gossip_in_nongroup_chat(self, acfactory, lp):
+        """
+        Test that we do not gossip peer keys in 1-to-1 chat,
+        as it makes no sense to gossip to peers their own keys.
+        """
+        ac1, ac2 = acfactory.get_two_online_accounts()
+
+        lp.sec("ac1: create chat with ac2")
+        chat = self.get_chat(ac1, ac2)
+
+        lp.sec("sending text message from ac1 to ac2")
+        msg_out = chat.send_text("message1")
+        assert not msg_out.is_encrypted()
+
+        lp.sec("wait for ac2 to receive message")
+        ev = ac2._evlogger.get_matching("DC_EVENT_MSGS_CHANGED")
+        assert ev[2] == msg_out.id
+        msg_in = ac2.get_message_by_id(msg_out.id)
+        assert msg_in.text == "message1"
+
+        lp.sec("create new chat with contact and send back (encrypted) message")
+        chat2b = ac2.create_chat_by_message(msg_in)
+        chat2b.send_text("message-back")
+
+        lp.sec("wait for ac1 to receive message")
+        ev = ac1._evlogger.get_matching("DC_EVENT_INCOMING_MSG")
+        assert ev[1] == chat.id
+        assert ev[2] > msg_out.id
+        msg_back = ac1.get_message_by_id(ev[2])
+        assert msg_back.text == "message-back"
+
+        # Gossip is only sent in encrypted messages,
+        # so we check that we actually sent an encrypted
+        # message.
+        assert msg_back.is_encrypted()
+        # No gossip is sent in 1-to-1 chat
+        assert chat2b.get_summary()["gossiped_timestamp"] == 0
```

Comment:
i think we have other tests that basically contain all of the checks except this line -- i'd just integrate it there with a comment.  Each such functional test takes 10-15 seconds and in this case the test is rather indirect anyway.  So it's fine to combine it with another test.

Reply:
Merged this check into `test_send_and_receive_will_encrypt_decrypt`